components/sections/query/custom-checkbox.js

```diff
@@ -0,0 +1,154 @@
+import { useState } from 'react'
+import { Box } from 'theme-ui'
+
+const sx = {
+  checkbox: {
+    stroke: 'secondary',
+    strokeWidth: '1.25',
+    width: '18px',
+    cursor: 'pointer',
+    transition: 'stroke 0.15s',
+    'input:not(:checked) ~ &': {
+      stroke: 'secondary',
+    },
+    '@media (hover: hover) and (pointer: fine)': {
+      'input:hover ~ &': { stroke: 'primary' },
+    },
+    'input:focus ~ &': {
+      bg: 'inherit',
+    },
+    'input:focus-visible ~ &': {
+      outline: 'dashed 1px rgb(110, 110, 110, 0.625)',
+      background: 'rgb(110, 110, 110, 0.625)',
+    },
+  },
+}
+
+const CheckboxIcon = ({
+  checkedIcon: CheckedIcon,
+  uncheckedIcon: UncheckedIcon,
+}) => {
+  return (
+    <>
+      <CheckedIcon
+        sx={{
+          ...sx.checkbox,
+          opacity: 0,
+          'input:checked ~ &': {
+            opacity: 1,
+            stroke: 'primary',
+          },
+        }}
+      />
+      <UncheckedIcon
+        sx={{
+          ...sx.checkbox,
+          position: 'absolute',
+          left: 0,
+          opacity: 1,
+          'input:checked ~ &': {
+            opacity: 0,
+            stroke: 'primary',
+          },
+        }}
+      />
+    </>
+  )
+}
+
+const CheckboxIconWithHover = ({
+  checkedIcon: CheckedIcon,
+  uncheckedIcon: UncheckedIcon,
+  checkedHoverIcon: CheckedIconHover,
+  uncheckedHoverIcon: UncheckedIconHover,
+}) => {
+  return (
+    <>
+      <CheckedIconHover
+        sx={{
+          ...sx.checkbox,
+          opacity: 0,
+          transition: 'stroke 0s',
+          'input:checked:hover ~ &': {
+            opacity: 1,
+            stroke: 'primary',
+          },
+        }}
+      />
+      <CheckedIcon
+        sx={{
+          ...sx.checkbox,
+          position: 'absolute',
+          left: 0,
+          opacity: 0,
+          'input:checked ~ &': {
+            opacity: 1,
+            stroke: 'primary',
+          },
+          'input:hover ~ &': {
+            opacity: 0,
+          },
+        }}
+      />
+      <UncheckedIcon
+        sx={{
+          ...sx.checkbox,
+          position: 'absolute',
+          left: 0,
+          opacity: 1,
+          'input:checked ~ &': {
+            opacity: 0,
+            stroke: 'primary',
+          },
+        }}
+      />
+    </>
+  )
+}
+
+const CustomCheckbox = ({
+  sx,
```

Comment:
Hm this is currently not being used -- should this be spread into the `input` element's styles?

Reply:
good catch! i intended what you say, and fixed it to do so. note that we're not using the custom sx at all here, and there are some nuances we'll need to document or revisit when we move this to `@carbonplan/components`, for example, as implemented now, changing the color would require respecifying colors for all the focus / check state modifies... anyway, we can tackle that another day!